whylog/assistant/regex_assistant/regex.py

```diff
@@ -0,0 +1,85 @@
+"""
+Regex verification and creating (but not finding groups in regex)
+"""
+
+import re
+
+from whylog.assistant.regex_assistant.exceptions import NotMatchingRegexError
+
+special_characters = frozenset('.^$*+{}?[]|()]')
+
+
+def verify_regex(regex, text):
+    """
+    Verifies regex properties such as:
+    - matching a whole text
+    - matching text in a one way only
+    If properties are not met, proper exceptions are returned.
+    """
+    # TODO: verify if regex matches text in a one way only
+
+    # regex must match a whole text from its beginning to end.
+    whole_text_regex = "^" + regex + "$"
+    pattern = re.compile(whole_text_regex)
+
+    match = re.match(pattern, text)
+    matched = False
+    groups = []
+    errors = []
+    if match is not None:
+        matched = True
+        groups = match.groups()
+    else:
+        errors.append(NotMatchingRegexError(text, regex))
+    return matched, groups, errors
```

Comment:
`matched` is always false when groups are empty and is always true when groups are not empty. It can be removed.